src/main/java/htsjdk/samtools/SAMTextHeaderCodec.java

```diff
@@ -323,11 +326,30 @@ private void reportErrorParsingLine(String reason, final SAMValidationError.Type
                             SAMValidationError.Type.HEADER_TAG_MULTIPLY_DEFINED, null);
                     continue;
                 }
+                validateSortOrderValue(keyAndValue);
                 mKeyValuePairs.put(keyAndValue[0], keyAndValue[1]);
             }
             lineValid = true;
         }
 
+        private void validateSortOrderValue(String[] value) {
```

Comment:
I think it would be slightly clearer if the caller extracted `value[0]` and `value[1]` so they could be declared as `String key` and `String value` here.